pkg/kubernetes/pkg/util/metrics/metrics.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"net/http"
+	"sync"
+
+	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var registerMetrics sync.Once
+
+// Metrics are identified in Prometheus by concatinating Namespace, Subsystem and Name
+// omitting any nulls and separating each with an underscore
+const (
+	CatalogNamespace = "servicecatalog"
+)
+
+var (
+	BrokerServiceClassCount = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: CatalogNamespace,
+			Name:      "broker_service_class_count",
+			Help:      "Number of services classes by Broker.",
+		},
+		[]string{"broker"},
```

Comment:
So, this metric is broken down by broker name?

Reply:
Yes, It's a vector of related metrics.  The grand total is referenced with the metric name (servicecatalog_broker_service_class_count) and is automatically broken down by broker name when graphed or you can explicitly reference the count for specific brokers  such as servicecatalog_broker_service_class_count{broker='ups-broker'}.